lib/spack/spack/cmd/uninstall.py

```diff
@@ -212,16 +214,23 @@ def do_uninstall(env, specs, force):
         if env:
             _remove_from_env(item, env)
 
-    # Sort packages to be uninstalled by the number of installed dependents
-    # This ensures we do things in the right order
-    def num_installed_deps(pkg):
-        dependents = spack.store.db.installed_relatives(
-            pkg.spec, 'parents', True)
-        return len(dependents)
+    # A package is ready to be uninstalled when nothing else references it,
+    # unless we are requested to force uninstall it.
+    is_ready = lambda x: not spack.store.db.query_by_spec_hash(x)[1].ref_count
+    if force:
+        is_ready = lambda x: True
+
+    while packages:
+        ready = [x for x in packages if is_ready(x.spec.dag_hash())]
+        if not ready:
+            msg = 'unexpected error [cannot proceed uninstalling specs with' \
```

Comment:
@alalazo everything looks good except there appears to be a typo here: is the `[` intentional?

Reply:
Yes, it is intentional, but I missed the closing square bracket. I think I used other times the format:
```
<short error message> [<details>]
```
let me know if that could be fine (I am not particularly attached to it if you think it should be changed).